app/src/main/java/com/infomaniak/mail/data/cache/mailboxInfo/SignatureController.kt

```diff
@@ -31,7 +31,7 @@ class SignatureController @Inject constructor(
 
     //region Get data
     fun getSignaturesAsync(mailboxObjectId: String): Flow<RealmList<Signature>> {
-        return mailboxController.getMailboxAsync(mailboxObjectId).mapNotNull { it.obj?.signatures }
+        return mailboxController.getMailboxAsync(mailboxObjectId).mapNotNull { it.obj?.local?.signatures }
```

Comment:
Is it worth it to query the whole mailbox when we only need the local values' object? (This pattern appears in other controllers as well)

Reply:
This is an EmbeddedObject, we can't query it directly.

Reply:
What, I asked @sirambd because I feared this exact situation and he told me we could. Maybe he did not understand what I meant

Reply:
We can query it, but not directly.
There is no PrimaryKey to search for since it's an EmbeddedObject.

Reply:
As long as we can query it indirectly by checking its parent's primary key, it should be enough. But I don't know to what extent this is feasible